apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/FissClaimResponseTransformerV2.java

```diff
@@ -88,50 +85,129 @@ private static ClaimResponse transformClaim(PreAdjFissClaim claimGroup) {
     ClaimResponse claim = new ClaimResponse();
 
     claim.setId("f-" + claimGroup.getDcn());
-    claim.setMeta(new Meta().setLastUpdated(Date.from(claimGroup.getLastUpdated())));
+    claim.setContained(List.of(getContainedPatient(claimGroup)));
     claim.setExtension(getExtension(claimGroup));
     claim.setIdentifier(getIdentifier(claimGroup));
     claim.setStatus(ClaimResponse.ClaimResponseStatus.ACTIVE);
     claim.setOutcome(STATUS_TO_OUTCOME.get(Character.toLowerCase(claimGroup.getCurrStatus())));
     claim.setType(getType());
     claim.setUse(ClaimResponse.Use.CLAIM);
-    claim.setCreated(new Date());
     claim.setInsurer(new Reference().setIdentifier(new Identifier().setValue("CMS")));
-    claim.setPatient(getPatient(claimGroup));
+    claim.setPatient(new Reference("#patient"));
     claim.setRequest(new Reference(String.format("Claim/f-%s", claimGroup.getDcn())));
 
+    FhirContext ctx = FhirContext.forR4();
+    IParser parser = ctx.newJsonParser();
+
+    String claimContent = parser.encodeResourceToString(claim);
+
+    String resourceHash = DigestUtils.sha1Hex(claimContent);
+
+    claim.setMeta(
+        new Meta()
+            .setLastUpdated(Date.from(claimGroup.getLastUpdated()))
+            .setVersionId("f-" + claimGroup.getDcn() + "-" + resourceHash));
+    claim.setCreated(new Date());
+
     return claim;
   }
 
+  private static Resource getContainedPatient(PreAdjFissClaim claimGroup) {
```

Comment:
would it be worth putting the Patient methods in a utility class so both Claim and ClaimResponse could reuse the logic?

Reply:
Yes I think it would, I realized it after the fact how redundant they were, when the 0.1+ updates pass I'll be changing them all to use a common method in this PR.